### PR TITLE
[fix][broker] Load loadAllResourceGroups with sync, and async in listener

### LINKED ISSUE #15113: ResourceGroupConfigListener and ResourceGroupNamespaceConfigListener has an cocurrent bug  at startrup
**Describe the bug**

```
 public ResourceGroupConfigListener(ResourceGroupService rgService, PulsarService pulsarService) {
        this.rgService = rgService;
        this.pulsarService = pulsarService;
        this.rgResources = pulsarService.getPulsarResources().getResourcegroupResources();
        loadAllResourceGroups(); // THIS IS ASYNC  1 
        this.rgResources.getStore().registerListener(this);
        rgNamespaceConfigListener = new ResourceGroupNamespaceConfigListener(
                rgService, pulsarService, this);
    }
```
 
        loadAllResourceGroups(); // THIS IS ASYNC  1 
 add invoke 
       resourceGroupsMap.put(rgName, rg);

```
    public ResourceGroupNamespaceConfigListener(ResourceGroupService rgService, PulsarService pulsarService,
            ResourceGroupConfigListener rgConfigListener) {
        this.rgService = rgService;
        this.pulsarService = pulsarService;
        this.namespaceResources = pulsarService.getPulsarResources().getNamespaceResources();
        this.tenantResources = pulsarService.getPulsarResources().getTenantResources();
        this.rgConfigListener = rgConfigListener;
        loadAllNamespaceResourceGroups(); // THIS IS ASYNC  2
        this.namespaceResources.getStore().registerListener(this);
    }

```
loadAllNamespaceResourceGroups(); // THIS IS ASYNC  2 

will invoke
                rgService.registerNameSpace(policy.resource_group_name, ns);

then
```
   private ResourceGroup checkResourceGroupExists(String rgName) throws PulsarAdminException {
        ResourceGroup rg = this.getResourceGroupInternal(rgName);
        if (rg == null) {
            throw new PulsarAdminException("Resource group does not exist: " + rgName);
        }
        return rg;
    }
```
inside
```
    /**
     * Get the RG with the given name. For internal operations only.
     */
    private ResourceGroup getResourceGroupInternal(String resourceGroupName) {
        if (resourceGroupName == null) {
            throw new IllegalArgumentException("Invalid null resource group name: " + resourceGroupName);
        }

        return resourceGroupsMap.get(resourceGroupName);
    }

```

if 2 run earlier than 1, it will occur error, if no resource group is added, it won't recover.

**To Reproduce**

review the code

**Expected behavior**

maybe we need loadAllResourceGroups sync, or when it done, then loadAllNamespaceResourceGroups

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**
master branch

### PR DESCRIPTION
**Pull request by leizhiyuan**
Fixes #15113

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #15113

### Motivation

fix the bug in #15113

### Modifications

load loadAllResourceGroups with sync, and async in listener

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [x] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java [hunk 1]
```diff
@@ -92,6 +95,14 @@ private void loadAllResourceGroups() {
         });
     }
 
+    private List<String> loadAllResourceGroups() {
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java:98]: It looks like we don't need to return the List here

**Review by eolivelli — CHANGES_REQUESTED** [review]
Overall LGTM
I left one comment PTAL

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java [hunk 2]
```diff
@@ -53,14 +56,14 @@ public ResourceGroupConfigListener(ResourceGroupService rgService, PulsarService
...
-        loadAllResourceGroups();
+        loadAllResourceGroups(); // ensure resource group config load completely.
         this.rgResources.getStore().registerListener(this);
         rgNamespaceConfigListener = new ResourceGroupNamespaceConfigListener(
                 rgService, pulsarService, this);
     }
 
-    private void loadAllResourceGroups() {
-        rgResources.listResourceGroupsAsync().whenCompleteAsync((rgList, ex) -> {
+    private CompletableFuture<List<String>> loadAllResourceGroupsAsync() {
```
Comment (tisonkun) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListener.java:65]: Perhaps we can chain a noop and return `CompletableFuture<Void>` here also.

**Review by tisonkun — APPROVED** [review]
LGTM. One minor comment.

**Comment by leizhiyuan** [comment]
ref https://github.com/apache/pulsar/pull/21781

**Pull request closed without merging**